### DISCUSSION TITLE
How to filter on a foreign table while not excluding the contents of the foreign table

### DISCUSSION THREAD

**Discussion by Strijdhagen** [supabase/supabase#10734] [Questions]
Hi all,

I'm hoping you can help me with a challenge that I'm facing.
I have two tables that are joined by a foreign key: Organisation and Cities.
For certain pages I want to display organisations that are based in a cerain city, for example New York.
The issue is, when I apply an .eq() filter, the values for all other cities for an org in New York will be NULL. 
I still want to display the other cities on that page.
Is there a way to filter on organisations, applying the filter on cities, but not actually filtering out the cities.

The code that I've been testing is the following in Next.JS:

```
.from("organisations")
    .select(
      "logo, name, homepage, cities(id, slug, name)"
    ).eq('cities.name', "New York")
```

**Comment by GaryAustin1** [comment]
You can do something like `".... , cities(id,slug,name), cities_filter:cities(name)"` then .eq('cities_filter.name',....).

You might need cities_filter:cities!inner(name).... not sure.

https://github.com/supabase/supabase/discussions/6587#discussioncomment-2657550

**Reply by Strijdhagen** [reply]
Thanks Gary, this works but I'm now facing an issue with optional data. For example, if I want to extend this to include and filter on 'States' which are in separate table. Only cities in the US have states in my dataset.
Adding !inner to 'States' (to filter for US pages) will filter out any organisations without states.
Is there a way to filter only on 'States' and ignore 'Cities' with NULL when using inner:

cities_filter:cities!inner(slug, states_filter:states!inner(slug))

**Reply by Strijdhagen** [reply]
In a similar vain I'm facing the same issue with a 'Industries" table. Not all organisations have industries. Applying an !inner on industries within the Select() query will filter out industries everywhere

**Reply by GaryAustin1** [reply]
Beyond my abilities.   I believe !inner will filter tables higher, while leaving it out will not.  May need aliases with two table references like the first thing I showed.
Here is main reference for this stuff: https://postgrest.org/en/stable/api.html#resource-embedding
Note that it is http url format, but the idea is the same for the js client code.
